app_flutter/lib/task_matrix.dart

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'package:cocoon_service/protos.dart' show CommitStatus, Stage, Task;
+
+/// Simplify [List<CommitStatus>] for easy retrieval of tasks and ordering of columns.
+///
+/// Construction performs the following steps:
+///   1. Scan through every [Task] in [List<CommitStatus>] to find all unique
+///      columns in the matrix. Uniqueness is defined by the key "[Task.stageName]:[Task.name]".
+///   2. Construct an empty [List<Column>] based on the size given from (1).
+///   3. Scan through [List<CommitStatus>] again, but insert every [Task] into
+///      their associated [Column] at the row they appear in.
```

Comment:
nit: what is "at the row they appear in."

Reply:
Reworded. Let me know if I need to clarify further.